### DISCUSSION TITLE
Overriding url loading

### DISCUSSION THREAD

**Discussion by farhazulMullick** [KevinnZou/compose-webview-multiplatform#79] [Ideas]
My requirement is if a web-content shows youtube url or twitter url then i want to redirect to respected apps. 
In native android i use `shouldOverrideUrlLoading` method in `android.webkitWebViewClient`. 

Does this library provide any alternate solution to this?

```
    public boolean shouldOverrideUrlLoading(WebView view, WebResourceRequest request) {
        return shouldOverrideUrlLoading(view, request.getUrl().toString());
    }
```

**Comment by KevinnZou** [comment]
@farhazulMullick Hi, thanks for your suggestion! I believe the question is the same as the problem we discussed in this [issue](https://github.com/KevinnZou/compose-webview-multiplatform/issues/37). I have completed the basic implementation for Android and iOS platforms, but I am facing some issues on the Desktop platform that need to be fixed by @DatL4g. Additionally, there is a problem with `shouldOverrideUrlLoading` not being able to intercept all requests. Therefore, we also need to find a solution for that. Feel free to provide your suggestions in the same issue. Thanks again!

**Comment by farhazulMullick** [comment]
Hey @KevinnZou Hello buddy.
Since i hand-picked whole project in my repo, now i can customise the code as per my requirements. 

Can you help/suggest me to add some kind of interceptor like `shouldOverrideUrlLoading` in ios?

**Reply by KevinnZou** [reply]
@farhazulMullick You could check this [branch](https://github.com/KevinnZou/compose-webview-multiplatform/tree/feature/support_request_interception).

**Reply by farhazulMullick** [reply]
Thanks kevinn! 
This library is really amazing. Saved lots of efforts.
